src/method/method_invoke_group.cpp

```diff
@@ -343,6 +343,12 @@ namespace cubmethod
   query_cursor *
   method_invoke_group::create_cursor (QUERY_ID query_id, bool oid_included)
   {
+    if (query_id == NULL_QUERY_ID || query_id >= SHRT_MAX)
```

Comment:
what's the reason for SHRT_MAX is used to check validation ?

Reply:
There is the same validation in query manager.